### PR TITLE
fix: body used already error fix #340

### LINKED ISSUE #340: Fetch adapter should not call body.text() if body.json() throws an error
If my geo data provider fails to give a healthy response, the error handling is not correct. I'm seeing errors of the following form

```
Error getting geography information for San Francisco, CA, USA
Error: HttpError: body used already for: https://nominatim.openstreetmap.org/search?addressdetails=1&q=San+Francisco%2C+CA%2C+USA&format=json
    at /home/styff/platform/node_modules/node-geocoder/lib/httpadapter/fetchadapter.js:58:15
    at tryCatcher (/home/stuff/platform/node_modules/bluebird/js/release/util.js:16:23)
```
    
 This seems to be the same as https://github.com/node-fetch/node-fetch/issues/533. The cause is this block in https://github.com/nchaulet/node-geocoder/blob/master/lib/httpadapter/fetchadapter.js:

```
        try {
          return await res.json();
        } catch (e) {
          throw new HttpError(await res.text(), {
            code: res.statusCode
          });
        }
```
  
  Because the response is a stream, calling `res.json()` and then `res.text()` is not valid. The recommended solution seems to be either:
- pre-check the response's status code before calling `res.json()` to reduce the likelihood of error
- just call `res.text()` so there's always a response assigned to a variable, and then call JSON.parse(`the-text`) and wrap that call in a try-catch. If the try-catch fails, then you can print `the-text` without trying to read the stream twice.

### PR DESCRIPTION
**Pull request by Ephigenia**
Fix "body used already for" error message as json and text is called. Should fix https://github.com/nchaulet/node-geocoder/issues/340

### CODE REVIEW

File: lib/httpadapter/fetchadapter.js [hunk 1]
```diff
@@ -42,10 +42,11 @@ class FetchAdapter {
         if (fullResponse) {
           return res;
         }
+        const rawResponseBody = res.text();
```
Comment (nchaulet) [inline lib/httpadapter/fetchadapter.js:45]: I think we need an `await` here

**Review by nchaulet — APPROVED** [review]

**Comment by nchaulet** [comment]
Thanks for the contribution I will publish a new version with that change later this week

**Comment by kamieniarz** [comment]
Hey @nchaulet, any chance you could release new version with this fix?

**Comment by nrathi** [comment]
+1, I don't think a new version was published alongside my fix a few months ago. I've still been using my patch.

https://github.com/nchaulet/node-geocoder/commit/e225f77b1c2e7d09930309b1e0ac53e15bd88552

**Comment by nrathi** [comment]
@nchaulet can you please publish a new version with my commit and the one from @kamieniarz ?

**Comment by nchaulet** [comment]
Just published a new version 4.4.0 with your changes

**Pull request merged**